src/Mailer/Transport/SmtpTransport.php

```diff
@@ -271,26 +271,42 @@ protected function _connect(): void
      */
     protected function _auth(): void
     {
-        if (isset($this->_config['username'], $this->_config['password'])) {
-            $replyCode = (string)$this->_smtpSend('AUTH LOGIN', '334|500|502|504');
-            if ($replyCode === '334') {
-                try {
-                    $this->_smtpSend(base64_encode($this->_config['username']), '334');
-                } catch (SocketException $e) {
-                    throw new SocketException('SMTP server did not accept the username.', null, $e);
-                }
-                try {
-                    $this->_smtpSend(base64_encode($this->_config['password']), '235');
-                } catch (SocketException $e) {
-                    throw new SocketException('SMTP server did not accept the password.', null, $e);
-                }
-            } elseif ($replyCode === '504') {
-                throw new SocketException('SMTP authentication method not allowed, check if SMTP server requires TLS.');
-            } else {
-                throw new SocketException(
-                    'AUTH command not recognized or not implemented, SMTP server may not require authentication.'
-                );
+        if (!isset($this->_config['username'], $this->_config['password'])) {
+            return;
+        }
+
+        $username = $this->_config['username'];
+        $password = $this->_config['password'];
+
+        $replyCode = $this->_smtpSend(
+            sprintf(
+                'AUTH PLAIN %s',
+                base64_encode($username . chr(0) . $username . chr(0) . $password)
+            ),
+            '235|504|535'
```

Comment:
Here I am checking for `504` and `535` to allow falling back to `AUTH LOGIN`. 

garas suggested checking `504 5.7.4 Unrecognized Authentication Type` but the ticket author mentioned his server returned `535 5.7.8 Error: authentication failed: Invalid authentication mechanism` when using `AUTH LOGIN` when it only supports `AUTH PLAIN`. So I am guess if the opposite was true then too a server could return `535` and not `504`.

Reply:
[RFC 4954 - SMTP Service Extension for Authentication](https://tools.ietf.org/html/rfc4954#section-6):

- 504 5.7.4 Unrecognized authentication type
- 534 5.7.9  Authentication mechanism is too weak
- 535 5.7.8  Authentication credentials invalid

It looks server should reply with 534, but replied with 535.

With wrong credentials, it should stop here and not retry `AUTH LOGIN`. But if there is non-compliant servers, then maybe retry if server responded `504|534|535`